internal/discovery/client.go

```diff
@@ -7,38 +7,38 @@ This class contains the behaviour of propagating a nodes status updates to disco
 import (
 	"context"
 	"fmt"
+	"time"
+
 	_ "github.com/Jille/grpc-multi-resolver"
 	"github.com/flipkart-incubator/dkv/internal/hlc"
 	"github.com/flipkart-incubator/dkv/pkg/serverpb"
+	"github.com/spf13/viper"
 	"go.uber.org/zap"
 	"google.golang.org/grpc"
-	"gopkg.in/ini.v1"
-	"strconv"
-	"time"
 )
 
 type DiscoveryClientConfig struct {
-	DiscoveryServiceAddr string
+	DiscoveryServiceAddr string `mapstructure:"discovery-service-addr"`
 	// time in seconds to push status updates to discovery server
-	PushStatusInterval time.Duration
+	PushStatusInterval time.Duration `mapstructure:"push-status-interval"`
 	// time in seconds to poll cluster info from discovery server
-	PollClusterInfoInterval time.Duration
+	PollClusterInfoInterval time.Duration `mapstructure:"poll-cluster-info-interval"`
+	//
 }
 
-func NewDiscoveryClientConfigFromIni(sect *ini.Section) (*DiscoveryClientConfig, error) {
-	sectConf := sect.KeysHash()
-	if discoveryServiceAddr, ok := sectConf["discoveryServiceAddr"]; ok {
-		if pushStatusInterval, err := strconv.Atoi(sectConf["pushStatusInterval"]); err == nil {
-			if pollClusterInfoInterval, err := strconv.Atoi(sectConf["pollClusterInfoInterval"]); err == nil {
-				return &DiscoveryClientConfig{
-					DiscoveryServiceAddr:    discoveryServiceAddr,
-					PushStatusInterval:      time.Duration(pushStatusInterval),
-					PollClusterInfoInterval: time.Duration(pollClusterInfoInterval),
-				}, nil
-			}
-		}
+func NewDiscoveryClientConfigFromYaml() (*DiscoveryClientConfig, error) {
+	viper.SetConfigName("dkv_config") // name of config file (without extension)
+	viper.SetConfigType("yaml")       // REQUIRED if the config file does not have the extension in the name
+	viper.AddConfigPath("./opts")
+	err := viper.ReadInConfig() // Find and read the config file
+	if err != nil {             // Handle errors reading the config file
+		fmt.Println("Fatal error config file: %w \n", err)
+	}
+	var DiscoveryClientConfig DiscoveryClientConfig
+	if err := viper.Unmarshal(&DiscoveryClientConfig); err != nil {
+		fmt.Println(err)
 	}
-	return nil, fmt.Errorf("Invalid discovery client configuration. Check section %s", sect.Name())
+	return &DiscoveryClientConfig, nil
```

Comment:
This should not be done. viper deserialization is already done , it should automatically handle an object of type `DiscoveryClientConfig` and avoid this step.